src/Compilers/Core/Portable/Operations/Loops/IForEachLoopStatement.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Semantics
+{
+    /// <summary>
+    /// Represents a C# 'foreach' statement or a VB 'For Each' staement.
+    /// </summary>
+    /// <remarks>
+    /// This interface is reserved for implementation by its associated APIs. We reserve the right to
+    /// change it in the future.
+    /// </remarks>
+    public interface IForEachLoopStatement : ILoopStatement
+    {
+        /// <summary>
+        /// Iteration variable of the loop.
+        /// </summary>
+        ILocalSymbol IterationVariable { get; }
```

Comment:
>IterationVariable [](start = 21, length = 17)

It feels confusing to have this property and the LoopControlVariable below. Especially that this property doesn't cover the case of deconstruction foreach in C#.

Reply:
What would be the recommended API here? We need the LoopControlVariable IOperation for VB, but that will always be null for C# and so we need the IterationVariable local symbol. We can also discuss it at the design meeting.

Reply:
LoopControlVariable should probably never be null. It is either:
- a declaration, or
- expression (handles VB case and C# deconstruction case)

Reply:
And I would remove the IterationVariable property.

Reply:
Should probably have Locals property as in IForLoopStatement